src/main/java/org/apache/commons/jxpath/ClassFunctions.java

```diff
@@ -91,6 +92,13 @@ public Function getFunction(
         final String namespace,
         final String name,
         Object[] parameters) {
+        JXPathFilter jxPathFilter = new JXPathFilter();
```

Comment:
Not sure, but I am a little bit worried that this can be quite limiting to developers. We force them to use system properties in order to allow some classes in jxpath functions. Also, it is not possible to change the filter per JXContext

Reply:
I think we should have some kind of interface `JXPathFilter` that will have all required methods (e.g. `isClassNameExposed(className)`) that will be used as the parameter of the overloaded method `Functions.getFunction(...)`. The overloaded method would create the `SystemPropertyJXPathFilter` implementation that uses system properties to specify filters. 

Then we would add field to JXPathContext that would hold the default JXPathFilter for given context (by default our system properties based filter). We would also add methods to change this filter, so that the developer can configure per-context filters. 

The `JXPathFilter` itself should be either interface with all the methods or just some filter-holder that would contain all the filters (interfaces JXPathClassFilter, JXPathXyzFilter, ...), so that we can pass it through the structure.

Reply:
Thanks for the proposal. I've added an interface `JXPathFilter` and an implementation `SystemPropertyJXPathFilter` based on the system property of the allow list. Moreover, I have added a new default method `getFunction` to the `Functions` interface that takes a filter as a parameter. The default implementation simply calls the `getFunction` without the filter and can be overridden by implemented classes. This guarantees binary compatibility as @garydgregory pointer out.